packages/protocol/tipmanager/testframework.go

```diff
@@ -107,34 +104,34 @@ func (t *TestFramework) setupEvents() {
 		t.scheduledBlocks.Set(block.ID(), scheduledBlock)
 		t.scheduledBlocksMutex.Unlock()
 
-		t.TipManager.AddTip(scheduledBlock)
-	}))
+		t.Instance.AddTip(scheduledBlock)
+	})
 
-	t.TipManager.Events.TipAdded.Hook(event.NewClosure(func(block *scheduler.Block) {
+	event.Hook(t.Instance.Events.TipAdded, func(block *scheduler.Block) {
 		if debug.GetEnabled() {
 			t.test.Logf("TIP ADDED: %s", block.ID())
 		}
 		atomic.AddUint32(&(t.tipAdded), 1)
-	}))
+	})
 
-	t.TipManager.Events.TipRemoved.Hook(event.NewClosure(func(block *scheduler.Block) {
+	event.Hook(t.Instance.Events.TipRemoved, func(block *scheduler.Block) {
 		if debug.GetEnabled() {
 			t.test.Logf("TIP REMOVED: %s", block.ID())
 		}
 		atomic.AddUint32(&(t.tipRemoved), 1)
-	}))
+	})
 
-	t.mockAcceptance.BlockAcceptedEvent.Attach(event.NewClosure(func(block *blockgadget.Block) {
-		require.NoError(t.test, t.engine.NotarizationManager.NotarizeAcceptedBlock(block.ModelsBlock))
-	}))
+	event.Hook(t.mockAcceptance.BlockAcceptedEvent, func(block *blockgadget.Block) {
+		require.NoError(t.test, t.Engine.NotarizationManager.NotarizeAcceptedBlock(block.ModelsBlock))
+	})
 
-	t.engine.NotarizationManager.Events.EpochCommitted.Attach(event.NewClosure(func(details *notarization.EpochCommittedDetails) {
-		t.TipManager.PromoteFutureTips(details.Commitment)
-	}))
+	event.Hook(t.Engine.NotarizationManager.Events.EpochCommitted, func(details *notarization.EpochCommittedDetails) {
+		t.Instance.PromoteFutureTips(details.Commitment)
+	})
 
-	t.engine.EvictionState.Events.EpochEvicted.Attach(event.NewClosure(func(index epoch.Index) {
-		t.TipManager.Evict(index)
-	}))
+	event.Hook(t.Engine.EvictionState.Events.EpochEvicted, func(index epoch.Index) {
```

Comment:
Is `event.Hook(` here expected? As it was `Attach` before.

Reply:
It is easier to hook inside the tests